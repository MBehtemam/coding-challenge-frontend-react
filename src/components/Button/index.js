import styled from 'styled-components';

const Button = styled.button`
  background-color: ${props => {
    if (props.btnStyle === 'default') {
      return '#f4f4f4';
    }
    if (props.btnStyle === 'primary') {
      return '#523ee8';
    }
    if (props.btnStyle === `danger`) {
      return `rgb(247, 56, 89)`;
    }
    return '#ffffff';
  }};
  border: 0;
  border-color: ${props => props.default && '#e0e0e1'};
  color: #fff;
  cursor: pointer;
  font-size: 16px;
  line-height: 0;
  outline: 0;
  padding: 15px 15px;
`;
export default Button;
