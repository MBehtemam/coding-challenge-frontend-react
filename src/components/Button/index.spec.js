import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import Button from '.';

describe('Test suits for Button component', () => {
  it('should match with snapshot', () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
