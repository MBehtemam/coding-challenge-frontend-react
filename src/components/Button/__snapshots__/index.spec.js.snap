// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suits for Button component should match with snapshot 1`] = `
.c0 {
  border: 0;
  border-color: #e0e0e1;
  color: #000000;
  cursor: pointer;
  font-size: 16px;
  line-height: 0;
  outline: 0;
  padding: 15px 15px;
}

<button
  className="c0"
/>
`;
